src/components/theme/Image/Image.jsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { getImageAttributes } from '@plone/volto/helpers';
+
+/**
+ * Image component
+ * @param {object | string} image - Plone image as object or url
+ * @param {string} alt - Alternative text for image
+ * @param {string} className - CSS class attribute
+ * @param {string} role - img role attribute
+ * @param {boolean} critical - whether to load the image
+ */
+const Image = ({
+  image,
+  alt = '',
+  className,
+  role = 'img',
+  critical = false,
+  ...imageProps
+}) => {
+  const { src, srcSet } = getImageAttributes(image);
+  const [isClient, setIsClient] = useState(false);
+  const [srcset, setSrcset] = useState(
+    critical && srcSet ? srcSet.join(', ') : null,
+  );
+
+  useEffect(() => {
+    setIsClient(true);
+    if (!srcset && srcSet?.length > 0) setSrcset(srcSet.join(', '));
+  }, [srcSet, srcSet?.length, srcset]);
+
+  return (
+    <>
+      <picture>
+        {srcset && (isClient || critical) && <source srcSet={srcset} />}
+        <img
+          src={src}
+          alt={alt}
+          className={className}
+          role={role}
+          loading="lazy"
+          style={{ width: '100%', objectFit: 'cover' }}
+          {...imageProps}
+        />
+      </picture>
+      {!critical && (
+        <noscript
+          dangerouslySetInnerHTML={{
+            __html: `
+              <img
+                src="${src}"
+                ${srcSet?.length && `srcset="${srcSet.join(', ')}"`}
+                alt="${alt}"
+                className="${className}"
+                role="${role}"
```

Comment:
In this HTML string I think that attribute values strings should be escaped based on the rules in [this SO answer](https://stackoverflow.com/a/9189067/258462).